Fix bazel build for 8437b7f5.

jyknight opened the pull request:
When using parse_headers validation, rpc_server.h fails to build, since we don't expose any of the RPC deps in the Bazel build.

So at least for now, just exclude it.

jhuber6 reviewed (approved):
Thanks, it's not really intended to be used internally anyways.

jyknight committed:
Fix bazel build for 8437b7f5.

When using parse_headers validation, rpc_server.h fails to build,
since we don't expose any of the RPC deps in the Bazel build.

So at least for now, just exclude it.
